### PR TITLE
Add support for uploading multiple tarballs to github release

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/builder/issues/74

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

File: Makefile [hunk 1]
```diff
@@ -81,11 +81,19 @@ ci-lint: $(SOURCE) VERSION
...
+	
+	# Build for all supported OSs
+	for OS in darwin linux; do \
+		rm $(PROJECT); \
+		GOOS=$$OS make $(PROJECT); \
+		mkdir -p bin-dist/$$OS; \
+		cp README.md bin-dist/$$OS/; \
+		cp LICENSE bin-dist/$$OS/; \
+		cp $(PROJECT) bin-dist/$$OS/; \
+		tar czf bin-dist/$(PROJECT).$(VERSION).$$OS.tar.gz -C bin-dist/$$OS .; \
```
Comment (zeisss) [inline Makefile]: we are using bin-dist here for two things: :A: preparation folder for what should go into the release tarball :B: storing the release tarball. I think we should make this two folders.

File: util/file.go [hunk 2]
```diff
@@ -8,25 +8,26 @@ import (
 	"github.com/juju/errgo"
 )
 
-// FindFileInDirectory loops over all files in directory, returning the filename,
-// and a pointer to the opened file for a file that matches the predicate.
-func FindFileInDirectory(directory string, predicate func(string) bool) (string, *os.File, error) {
-	var filename string
+// FindFilesInDirectory finds all files in directory whose name matches the predicate,
+// and opens and returns them.
+func FindFilesInDirectory(directory string, predicate func(string) bool) ([]*os.File, error) {
```
Comment (zeisss) [inline util/file.go]: Can't we just replace this with Glob? https://golang.org/pkg/path/filepath/#Glob

Comment (JosephSalisbury) [inline util/file.go]: Oh yes we can, that's cool - til about filepath

File: Makefile (hunk 1 above)
Comment (JosephSalisbury) [inline Makefile]: sounds good to me

Comment (JosephSalisbury) [inline Makefile]: we now have:

```
$ ls -R ./build
total 0
drwxr-xr-x  5 joseph  staff   170B 31 Mar 16:02 darwin/
drwxr-xr-x  5 joseph  staff   170B 31 Mar 16:03 linux/

./build/darwin:
total 26256
-rw-r--r--  1 joseph  staff   552B 31 Mar 16:02 LICENSE
-rw-r--r--  1 joseph  staff   5.7K 31 Mar 16:02 README.md
-rwxr-xr-x  1 joseph  staff    13M 31 Mar 16:02 builder*

./build/linux:
total 26584
-rw-r--r--  1 joseph  staff   552B 31 Mar 16:03 LICENSE
-rw-r--r--  1 joseph  staff   5.7K 31 Mar 16:03 README.md
-rwxr-xr-x  1 joseph  staff    13M 31 Mar 16:03 builder*
joseph on Josephs-MacBook-Pro at 16:03:48 in ~/Repos/builder (multiple-tarballs)
$ ls -R ./bin-dist/
total 14792
-rw-r--r--  1 joseph  staff   3.6M 31 Mar 16:02 builder.0.13.3+git.darwin.tar.gz
-rw-r--r--  1 joseph  staff   3.6M 31 Mar 16:03 builder.0.13.3+git.linux.tar.gz
```

**Comment by Nesurion** [comment]
LGTM

**Pull request merged**